modules/standard/IO.chpl

```diff
@@ -2249,13 +2249,13 @@ proc open(out error:syserr, path:string="", mode:iomode, hints:iohints=IOHINT_NO
     var hostidx_end = new_str.find(":");
     var host = new_str[0..hostidx_end-1];
 
-    new_str = new_str[hostidx_end+1..new_str.length];
+    new_str = new_str[hostidx_end..new_str.length];
 
     var portidx_end = new_str.find("/");
     var port = new_str[0..portidx_end-1];
 
     //the file path is whatever we have left
-    var file_path = new_str[portidx_end+1..new_str.length];
+    var file_path = new_str.substring(portidx_end..new_str.length);
```

Comment:
substring doesn't exist anymore, and there seem to be other problems with this code.

Might I recommend you replace it with:


    // hdfs://<host>:<port>/<path>
    var host_start = path.find("//") + 2;
    var colon = path.find(":", host_start..);
    var slash = path.find("/", host_start..);
    var host_end, port_start, port_end, path_start = 0;

    if colon > 0 {
      host_end = colon - 1;
      port_start = colon + 1;
      if slash > 0 {
        port_end = slash - 1;
        path_start = slash + 1;
      } else {
        port_end = path.length;
      }
    } else {
      if slash > 0 {
        host_end = slash - 1;
        path_start = slash + 1;
      } else {
        host_end = path.length;
      }
    }
    var host = path[host_start..host_end];
    var port = "";
    if port_start > 0 then port = path[port_start..port_end];
    var file_path = "";
    if path_start > 0 then file_path = path[path_start..];


which works for me for the following calls:

    parse_hdfs_path("hdfs://<host>:<port>/<path>");
    parse_hdfs_path("hdfs://<host>/<path>");
    parse_hdfs_path("hdfs://<host>:<port>");

Eventually, we should move this parse function to something that can be independently tested.